services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/dataaccess/DataAccessRequestManagerImpl.java

```diff
@@ -0,0 +1,151 @@
+	}
+
+	@Override
+	public DataAccessRequestInterface getDataAccessRequestForUpdate(UserInfo userInfo, String accessRequirementId)
+			throws NotFoundException {
+		try {
+			DataAccessRequestInterface current = getUserOwnCurrentRequest(userInfo, accessRequirementId);
+			if (current instanceof DataAccessRenewal) {
+				return current;
+			}
+			ACTAccessRequirement requirement = (ACTAccessRequirement) accessRequirementDao.get(accessRequirementId);
+			if (requirement.getIsAnnualReviewRequired() /*TODO: && has approved submission*/) {
+				return createRenewalFromRequest(current);
+			}
+			return current;
+		} catch (NotFoundException e) {
+			return new DataAccessRequest();
+		}
+	}
+
+	public DataAccessRenewal createRenewalFromRequest(DataAccessRequestInterface current) {
+		DataAccessRenewal renewal = new DataAccessRenewal();
+		renewal.setId(current.getId());
+		renewal.setAccessRequirementId(current.getAccessRequirementId());
+		renewal.setResearchProjectId(current.getResearchProjectId());
+		renewal.setCreatedBy(current.getCreatedBy());
+		renewal.setCreatedOn(current.getCreatedOn());
+		renewal.setModifiedBy(current.getModifiedBy());
+		renewal.setModifiedOn(current.getModifiedOn());
+		renewal.setAccessors(current.getAccessors());
+		renewal.setAttachments(current.getAttachments());
+		renewal.setDucFileHandleId(current.getDucFileHandleId());
+		renewal.setIrbFileHandleId(current.getIrbFileHandleId());
+		renewal.setEtag(current.getEtag());
+		renewal.setConcreteType(DataAccessRenewal.class.getName());
+		return renewal;
+	}
+
+	@WriteTransactionReadCommitted
+	@Override
+	public DataAccessRequestInterface update(UserInfo userInfo, DataAccessRequestInterface toUpdate)
+			throws NotFoundException, UnauthorizedException {
+		ValidateArgument.required(userInfo, "userInfo");
+		validateRequest(toUpdate);
+
+		DataAccessRequestInterface original = dataAccessRequestDao.getForUpdate(toUpdate.getId());
+
+		if (!original.getEtag().equals(toUpdate.getEtag())) {
+			throw new ConflictingUpdateException("etag does not match.");
+		}
+
+		ValidateArgument.requirement(toUpdate.getCreatedBy().equals(original.getCreatedBy())
+				&& toUpdate.getCreatedOn().equals(original.getCreatedOn())
+				&& toUpdate.getAccessRequirementId().equals(original.getAccessRequirementId())
+				&& toUpdate.getResearchProjectId().equals(original.getResearchProjectId()),
+				"researchProjectId, accessRequirementId, createdOn and createdBy fields cannot be editted.");
+
+		if (toUpdate instanceof DataAccessRenewal) {
+			ACTAccessRequirement requirement = (ACTAccessRequirement) accessRequirementDao.get(toUpdate.getAccessRequirementId());
+			ValidateArgument.requirement(requirement.getIsAnnualReviewRequired()
+					/* && has approved submission */,
+					"Can only update to a DataAccessRenewal after a submission is approved and the requirement requires renewal.");
+		}
+		String researchProjectOwnerId = researchProjectDao.getOwnerId(toUpdate.getResearchProjectId());
```

Comment:
need more cases for requests with submissions.
if there is a submission, then it must be approved before any changes can be made regardless of the type.
if there is an approved submission, and and requirement requires renewal, then must be a renewal.